concepts/views.py

```diff
@@ -59,6 +59,17 @@ def merge_concepts(request, source_concept_id):
 
     return HttpResponseRedirect(next_page)
 
+@login_required
+def flag_concept(request, source_concept_id):
+    concept = get_object_or_404(Concept, pk=source_concept_id)
+    if request.method == "POST":
+        comment = request.POST.get("flag_comment", "").strip()
+        concept.concept_state = Concept.FLAGGED    
+        concept.comment = comment
+        concept.save()
+        next_page = request.GET.get('next', reverse('concepts'))
+    return HttpResponseRedirect(next_page)
```

Comment:
if request is not `POST` this will throw an exception since `next_page` does not exist.